tests/integration/vpp/040_acl_test.go

```diff
@@ -0,0 +1,993 @@
+//  Copyright (c) 2019 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package vpp
+
+import (
+	"github.com/ligato/cn-infra/logging/logrus"
+	acl "github.com/ligato/vpp-agent/api/models/vpp/acl"
+	. "github.com/onsi/gomega"
+	"strings"
+	"testing"
+
+	_ "github.com/ligato/vpp-agent/plugins/vpp/aclplugin"
+	aclplugin_vppcalls "github.com/ligato/vpp-agent/plugins/vpp/aclplugin/vppcalls"
+	_ "github.com/ligato/vpp-agent/plugins/vpp/ifplugin"
+	"github.com/ligato/vpp-agent/plugins/vpp/ifplugin/ifaceidx"
+	ifplugin_vppcalls "github.com/ligato/vpp-agent/plugins/vpp/ifplugin/vppcalls"
+	//	"encoding/json"
+)
+
+var aclNoRules []*acl.ACL_Rule
+
+var aclErr1Rules = []*acl.ACL_Rule{
```

Comment:
It's not that safe to define global variables like this and also takes a lot of space. 

It would be better to create a helper function that returns ACL rule, something like:

```go
func newAclIpRule(src, dst string) *acl.ACL_Rule {
	return &acl.ACL_Rule{
		Action: acl.ACL_Rule_PERMIT,
		IpRule: &acl.ACL_Rule_IpRule{
			Ip: &acl.ACL_Rule_IpRule_Ip{
				SourceNetwork:      src,
				DestinationNetwork: dst,
			},
		},
	}
}

func Test(...) {
	aclRules := []*acl.ACL_Rule{
		newAclIpRule(".0.", "10.20.0.0/24"),
	}
}
```

Reply:
so far partly done on next commit
still to need to do for denyICMP denyTCP denyUDP and such...